cross/libssh2/Makefile

```diff
@@ -15,5 +15,14 @@ CMAKE_ARGS += -DBUILD_SHARED_LIBS=ON
 CMAKE_ARGS += -DBUILD_STATIC_LIBS=OFF
 CMAKE_ARGS += -DBUILD_EXAMPLES=OFF
 CMAKE_ARGS += -DBUILD_TESTING=OFF
+ifneq ($(wildcard $(OPENSSL_STAGING_PREFIX)),)
+CMAKE_ARGS += -DOPENSSL_INCLUDE_DIR=$(OPENSSL_STAGING_PREFIX)/include
+CMAKE_ARGS += -DOPENSSL_SSL_LIBRARY=$(OPENSSL_STAGING_PREFIX)/lib/libssl.so
+CMAKE_ARGS += -DOPENSSL_CRYPTO_LIBRARY=$(OPENSSL_STAGING_PREFIX)/lib/libcrypto.so
+else
+CMAKE_ARGS += -DOPENSSL_INCLUDE_DIR=$(STAGING_INSTALL_PREFIX)/include
+CMAKE_ARGS += -DOPENSSL_SSL_LIBRARY=$(STAGING_INSTALL_PREFIX)/lib/libssl.so
+CMAKE_ARGS += -DOPENSSL_CRYPTO_LIBRARY=$(STAGING_INSTALL_PREFIX)/lib/libcrypto.so
+endif
```

Comment:
@hgy59 there's only a few cases where this is needed (`mariadb-connector-c`, `znc` and now `libssh2`) although wonder if this would not be worth adding to the cmake definition file to avoid case-by-case declarations when used from `spksrc.python.mk`.... Also to note, I'll move this in its own PR.